src/rpdk/core/contract/resource_client.py

```diff
@@ -201,6 +215,78 @@ def get_metadata(self):
                 and properties[prop]["insertionOrder"] == "false"
             }
 
+    def transform_model(self, input_model, output_model):
+        if self.property_transform_keys:
+            self.check_npm()
+        for prop in self.property_transform_keys:
+            document_input, _path_input, _parent_input = traverse(
+                input_model, list(prop)[1:]
+            )
+            document_output, _path_output, _parent_output = traverse(
+                output_model, list(prop)[1:]
+            )
+            if document_input != document_output:
+                transformed_property = self.transform(prop, input_model)
+                self.update_transformed_property(
+                    prop, transformed_property, input_model
+                )
+
+    def transform(self, property_path, input_model):
+
+        path = "/" + "/".join(property_path)
+        property_transform_value = self.property_transform[path].replace('"', '\\"')
+
+        content = self.transformation_template.render(
+            input_model=input_model, jsonata_expression=property_transform_value
+        )
+
+        file = tempfile.NamedTemporaryFile(
+            mode="w+b",
+            buffering=-1,
+            encoding=None,
+            newline=None,
+            suffix=".js",
+            prefix=None,
+            dir=".",
+            delete=True,
+        )
+        file.write(str.encode(content))
+
+        LOG.debug("Jsonata transformation content %s", file.read().decode())
+        jsonata_output = subprocess.getoutput("node " + file.name)
+
+        file.close()
+        return jsonata_output
+
+    # removing coverage for this method as it is not possible to check both npm
+    # exists and does not exist condition in unit test
+    @staticmethod
+    def check_npm():  # pragma: no cover
+        output = subprocess.getoutput("npm list jsonata")
+        if "npm: command not found" not in output:
+            if "jsonata@" not in output:
+                subprocess.getoutput("npm install jsonata")
+        else:
+            LOG.error(
```

Comment:
should we make this method return something? if a user doesn't have npm installed since this method doesn't return anything looks like we will just continue to run the transformation function.